File: codegen/search_solver_wrapper.scala
import scala.io.Source

object Main {
    def main(args: Array[String]) = {
        val qdimacs = Source.fromFile(args(0)).mkString
        val sat = SolverExport.run_search_solver(qdimacs)
        if (sat) System.exit(10) else System.exit(20)
    }
}

File: codegen/search_solver_wrapper.hs
module Main where
import SolverExport
import System.Exit
import System.Environment

main :: IO ()
main = do
  (input_filename:rest) <- getArgs
  qdimacs <- readFile input_filename
  let sat = run_search_solver qdimacs
  if sat then exitWith (ExitFailure 10) else exitWith (ExitFailure 20)
